packages/block-editor/src/store/reducer.js

```diff
@@ -743,6 +743,57 @@ const withResetControlledBlocks = ( reducer ) => ( state, action ) => {
 	return reducer( state, action );
 };
 
+/**
+ * Higher-order reducer that auto insert blocks.
+ *
+ * @param {Function} reducer Original reducer function.
+ *
+ * @return {Function} Enhanced reducer function.
+ */
+const withAutoInsertBlocks = ( reducer ) => ( state, action ) => {
```

Comment:
I have no idea at this stage. I'm trying to build a prototype that works with all nested editors and properly integrates with the undo/redo mechanism 😅

Reply:
If we were able to modify the initial parsing to have all auto-inserted blocks in place as discussed in https://github.com/WordPress/gutenberg/pull/49789#discussion_r1171377177 then we won't need this part at all. In fact, it would be as simple as extending the action creator to store the list of auto-inserted block names, so we can store them later in the database together with other changes to the edited content. This way, the next time we load the same content, we can skip the auto-inserting of blocks listed earlier because they are now controlled by the user.